src/client/redirect-watch.js

```diff
@@ -8,69 +8,84 @@ import domProcessor from './dom-processor/index';
 const HASH_RE = /#.*$/;
 
 export default class RedirectWatch extends EventEmiter {
-    constructor (codeInstrumentation, eventSandbox) {
+    constructor (listeners, codeInstrumentation, elementSandbox) {
         super();
 
         this.DETECTED_EVENT = 'hammerhead|event|redirect-detected';
 
```

Comment:
I have no problem with 
`redirectWatch.REDIRECT_DETECTED_EVENT`
otherwise (`redirectWatch.DETECTED_EVENT`) it looks like the watch itself is detected (a good watch shall never be detected!)